libs/api/ebpf_api.cpp

```diff
@@ -2323,22 +2322,32 @@ _initialize_ebpf_object_from_file(
     EBPF_LOG_ENTRY();
     ebpf_result_t result = EBPF_SUCCESS;
 
-    new_object->file_name = cxplat_duplicate_string(path);
+    if (std::holds_alternative<std::string>(file_or_data)) {
```

Comment:
nit: should we save result of `std::holds_alternative<std::string>(file_or_data)` in a variable so that can be used later in this function too?

Reply:
Not sure it's worth the complexity.